picker/picker.go

```diff
@@ -105,11 +105,16 @@ func (mwr *remotesWeightedRandom) Select(excludes ...string) (api.Peer, error) {
 	cum := 0.0
 	// calculate CDF over weights
 	for peer, weight := range mwr.remotes {
+		find:=false
 		for _, exclude := range excludes {
 			if peer.NodeID == exclude || peer.Addr == exclude {
-				continue
+				find=true
+				break
 			}
 		}
+		if !find{
+			continue
```

Comment:
Also you should 

1. use `gofmt -s <file>` to format your code. 
2. add `-s` flag to your git commit to sign. such as `git commit -s -m "git message"`

Otherwise CI will fail.

If you want to re-commit, you could also use `--amend` to squash multiple commit into a single one. So here is the workflow if you want to continue working on this PR:

1. follow @stevvooe 's instructions and update your code
2. use `gofmt -s <filename>` to format your code
3. `git commit -as -m "xxx"` to update latest commit
4. `git push -f xxx xxx` to update this PR

Or you can simply transit this commit to me, and let me to send a PR :)